images/base/Dockerfile

```diff
@@ -2,6 +2,23 @@ FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
 
 USER root
 
-RUN microdnf update \
-    && microdnf --setopt=install_weak_deps=0 --setopt=tsflags=nodocs install java-17-openjdk-headless shadow-utils \
-    && microdnf clean all
\ No newline at end of file
+ARG OAUTH_LIB_VERSION=0.15.0
+ARG NIMBUS_JOSE_JWT_VERSION=9.37.2
+
+# Install Java and other necessary packages
+RUN microdnf update -y && \
+    microdnf --setopt=install_weak_deps=0 --setopt=tsflags=nodocs install -y \
+    java-17-openjdk-headless shadow-utils wget && \
+    microdnf clean all
+
+# Download OAuth libraries
+RUN echo "Including OAuth libraries in the base image..." && \
+    mkdir -p /opt/kafka/oauth-libs && \
+    OAUTH_COMMON_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-common/${OAUTH_LIB_VERSION}/kafka-oauth-common-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_SERVER_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-server/${OAUTH_LIB_VERSION}/kafka-oauth-server-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_SERVER_PLAIN_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-server-plain/${OAUTH_LIB_VERSION}/kafka-oauth-server-plain-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_KEYCLOAK_AUTHORIZER_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-keycloak-authorizer/${OAUTH_LIB_VERSION}/kafka-oauth-keycloak-authorizer-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_CLIENT_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-client/${OAUTH_LIB_VERSION}/kafka-oauth-client-${OAUTH_LIB_VERSION}.jar" && \
+    NIMBUS_JOSE_JWT_URL="https://repo1.maven.org/maven2/com/nimbusds/nimbus-jose-jwt/${NIMBUS_JOSE_JWT_VERSION}/nimbus-jose-jwt-${NIMBUS_JOSE_JWT_VERSION}.jar" && \
```

Comment:
So if you wanna do it this way ... aren't you missing Jackson Databind and Json Path?

Reply:
Hmm, I tried using OAuth in test containers, and it worked well without any issues. If those libs were needed, would I get an error?

Reply:
Maybe a question on @mstruk?

Reply:
But I guess they might be needed only in some configs?

Reply:
> But I guess they might be needed only in some configs?
That's what I am thinking also 🤔 .

Reply:
Okay using OAUTHBEARER method throws
```
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonKey
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 53 more
```
when using producer and consumer so I think those dependencies are needed at least `Jackson Databind`